word_counter.rb

```diff
@@ -0,0 +1,18 @@
+class WordCounter
+  def self.count file  
+    raise 'WordCounter: no file given' unless file 
+    # use inject to create a hash of word counts
+    to_word_array(file).inject({}) { |h,w| h[w] ? h[w] += 1 : h[w] = 1; h }
```

Comment:
I hear you.  But multi-line chaining is generally frowned upon for readability reasons.  The solution is to turn the complicated parts into separate methods; then you can make a nice, brief chain that includes that method.